### PR TITLE
feat: Fruity Blood Overdrive

### PR DESCRIPTION
**Pull request by ttaschke**
* Adds a parser for `Fruity Blood Overdrive`
* Adds `.fst` file for testing `Fruity Blood Overdrive` parse testing

Screen Shot 2023-02-25 at 19 36 39

### CODE REVIEW

**Comment by ttaschke** [comment]
@demberto 
This is a version with `construct` and the changes discussed in the draft PR.

I made the `plugin_marker` optional for now, as I was not able to get a `.fst` from an older plugin version. Also, I am still unsure how this would look like on a global (`StructEventBase`?) layer. Please let me know your thoughts on this.
https://github.com/demberto/PyFLP/pull/120/files#diff-7cbba45382f7949d6cb914fe5ee6fea2836260673dc3f868c15f2ae7ea0ce383R94

File: pyflp/plugin.py [hunk 1]
```diff
@@ -88,6 +89,20 @@ class FruityBalanceEvent(StructEventBase):
     STRUCT = c.Struct("pan" / c.Int32ul, "volume" / c.Int32ul).compile()
 
 
+class FruityBloodOverdriveEvent(StructEventBase):
+    STRUCT = c.Struct(
+        "plugin_marker" / c.Optional(c.Bytes(4)),  # redesigned native plugin marker
```
Comment (demberto) [inline pyflp/plugin.py]: I have slightly changed the handling for `StructEventBase` in https://github.com/demberto/PyFLP/commit/bd493c2ac363af0efe768e7582de11b17ee9d265

A kwarg named `len` is passed to every construct by StructEventBase's ctor.

Use `c.If(c.this._.len == 33, c.Bytes(4))`. Also are you sure `x100` is just a single byte? I think Blood Overdrive's event was 36 bytes.
If it is, change the comparsion above to `c.this._.len == 36`.

File: pyflp/plugin.py [hunk 2]
```diff
@@ -750,6 +765,68 @@ class FruityBalance(_PluginBase[FruityBalanceEvent], _IPlugin, ModelReprMixin):
...
 
+class FruityBloodOverdrive(
+    _PluginBase[FruityBloodOverdriveEvent], _IPlugin, ModelReprMixin
+):
+    """![]()"""  # noqa
+
+    INTERNAL_NAME = "Fruity Blood Overdrive"
+
+    pre_band = _NativePluginProp[int]()
+    """Linear.
```
Comment (demberto) [inline pyflp/plugin.py:780]: Please keep an empty line between "Linear." and the table. It causes issues with sphinx-autodoc.

**Review by demberto — CHANGES_REQUESTED** [review]

File: pyflp/plugin.py (hunk 2 above)
Comment (ttaschke) [reply pyflp/plugin.py:780]: Fixed: a4ddc7e40ac9eba27ef6d43ef1fdef88018399f5

File: pyflp/plugin.py (hunk 1 above)
Comment (ttaschke) [reply pyflp/plugin.py]: Yes, you are right, `x100` should be 4 bytes and the total length 36 bytes.
Fixed in 1234acaf9eaeddd779d505ab0299672fdfa6dcfb.

**Comment by demberto** [comment]
Perfect :)

**Pull request merged**